src/main/java/org/jabref/model/groups/PopularityGroup.java

```diff
@@ -0,0 +1,146 @@
+        return true;
+    }
+
+    @Override
+    public boolean equals(Object o) {
+        if (this == o) {
+            return true;
+        }
+        if (!(o instanceof PopularityGroup other)) {
+            return false;
+        }
+        return Objects.equals(getName(), other.getName())
+                && Objects.equals(getHierarchicalContext(), other.getHierarchicalContext());
+    }
+
+    @Override
+    public int hashCode() {
+        return Objects.hash(getName(), getHierarchicalContext());
+    }
+
+    /**
+     * Retrieves the view count for a given BibEntry.
+     *
+     * @param entry The BibEntry whose view count is to be fetched.
+     * @return The number of views for the BibEntry, or 0 if not found.
+     */
+    public static int getEntryViewCount(BibEntry entry) {
+        MVMap<String, Integer> viewCounts = mvStore.openMap("entryViewCounts");
+        String uniqueKey = getUniqueKeyForEntry(entry);
+        return viewCounts.getOrDefault(uniqueKey, 0);
+    }
+
+    public static String getUniqueKeyForEntry(BibEntry entry) {
+        return entry.getField(StandardField.DOI).orElse(
+                entry.getField(StandardField.KEY).orElse(
+                        String.valueOf(entry.hashCode())
+                )
+        );
+    }
+
+    public static synchronized MVStore getMVStore() {
+        if (mvStore == null) {
+            Path mvStorePath = getOtherDataDir().resolve("tracking.mv");
+            mvStore = new MVStore.Builder().fileName(mvStorePath.toString()).open();
+        }
+        return mvStore;
+    }
+
+    public List<BibEntry> processPopularityGroup(List<BibEntry> allEntries, String timePeriodCombo, Integer maxEntriesCombo) {
+     List<BibEntry> filteredEntries = new ArrayList<>();
+        long currentTime = System.currentTimeMillis();
+        long timeThreshold = getTimeThreshold(currentTime, timePeriodCombo);
+
+        for (BibEntry entry : allEntries) {
+            long lastViewTime = getLastViewTime(entry);
+            if (lastViewTime >= timeThreshold) {
+                filteredEntries.add(entry);
+            }
+        }
+
+        filteredEntries.sort((e1, e2) -> Integer.compare(PopularityGroup.getEntryViewCount(e2), PopularityGroup.getEntryViewCount(e1)));
+
+        if (filteredEntries.size() > maxEntriesCombo) {
+            filteredEntries = filteredEntries.subList(0, maxEntriesCombo);
+        }
+
+        return filteredEntries;
+    }
+
+    private long getTimeThreshold(long currentTime, String selectedTimePeriod) {
+        return switch (selectedTimePeriod) {
+            case "Last week" ->
+                    currentTime - 7 * 24 * 60 * 60 * 1000L;
+            case "Last month" ->
+                    currentTime - 30 * 24 * 60 * 60 * 1000L;
+            case "Last year" ->
+                    currentTime - 365 * 24 * 60 * 60 * 1000L;
+            default -> // "All time"
+                    0;
+        };
+    }
```

Comment:
The factor should be stored upon group creation (I think on change of group properties, the group is re-created)

Always reading the property seems to slow down things.